src/components/Screen.tsx

```diff
@@ -61,11 +65,17 @@ export class InnerScreen extends React.Component<ScreenProps> {
       sheetGrabberVisible = false,
       sheetCornerRadius = -1.0,
       sheetExpandsWhenScrolledToEdge = true,
+      stackPresentation,
     } = rest;
 
     if (enabled && isNativePlatformSupported) {
-      AnimatedNativeScreen =
-        AnimatedNativeScreen || Animated.createAnimatedComponent(NativeScreen);
+      const AnimatedScreen =
+        Platform.OS === 'android' ||
+        stackPresentation === 'push' ||
+        stackPresentation === 'containedModal' ||
+        stackPresentation === 'containedTransparentModal'
+          ? AnimatedNativeScreen
+          : AnimatedNativeModalScreen;
```

Comment:
I would add some comment here, that because how iOS modals are being mounted in view hierarchy, we need to use separate component. Also, in the future we need to check if for sure Android modals are being mounted in the same view group as usual screens (and not in the separate view, same as on iOS).

Reply:
> Also, in the future we need to check if for sure Android modals are being mounted in the same view group as usual screens (and not in the separate view, same as on iOS).

You mean in the future as when the new modals will be merged or do you mean the current situation? Currently, modals on `Android` are just transparent fragments with the previous ones not being detached so they are in the same view hierarchy.

Reply:
I mean the modals, that will be introduced in Screens, since they may be pushed into the separate dialog hierarchy, but we'll check that, when the modal PR will be opened